src/BudgetManager/BudgetManager.CQRS/Handlers/TransactionHandlers/UpdateIncomeTransactionHandler.cs

```diff
@@ -13,41 +14,78 @@ public class UpdateIncomeTransactionHandler : IRequestHandler<UpdateIncomeTransa
+        public UpdateIncomeTransactionHandler(IMapper mapper, IMediator mediator, ITransactionRepository transactionRepository, IBaseRepository<User> userRepository)
         {
             _mapper = mapper;
             _mediator = mediator;
-            _dataAccess = dataAccess;
+            _transactionRepository = transactionRepository;
+            _userRepository = userRepository;
         }
 
         public async Task<IncomeTransactionResponse> Handle(UpdateIncomeTransactionCommand request, CancellationToken cancellationToken)
         {
-            var filter = Builders<Transaction>.Filter.And(
+            var transactionFilter = Builders<Transaction>.Filter.And(
                 Builders<Transaction>.Filter.Eq(t => t.UserId, request.userId),
                 Builders<Transaction>.Filter.Eq(t => t.Id, request.updateIncomeDTO.Id),
                 Builders<Transaction>.Filter.Eq(t => t.TransactionType, Model.Enums.OperationType.Income));
 
-            var projection = Builders<Transaction>.Projection.Include(t => t.WalletId).Include(t => t.Value);
-            var oldIncomeTransaction = (await _dataAccess.FilterBy<Transaction>(filter, projection, cancellationToken)).FirstOrDefault();
-            if (oldIncomeTransaction == null) { throw new KeyNotFoundException("Transaction not found"); }
+            var oldIncomeTransaction = (await _transactionRepository.FilterBy(transactionFilter, cancellationToken))
+                                .FirstOrDefault() ?? throw new KeyNotFoundException("Transaction not found.");
             bool walletChanged = oldIncomeTransaction.WalletId != request.updateIncomeDTO.WalletId;
 
+            var user = await _userRepository.FindByIdAsync(request.userId, cancellationToken);
+            var wallets = user.Wallets ?? throw new KeyNotFoundException("User has no wallets");
+            var updatedWallets = new List<Wallet>();
+
+            if (walletChanged)
+            {
+                foreach (var wallet in user.Wallets)
+                {
+                    if (wallet.Id == oldIncomeTransaction.WalletId)
+                    {
+                        wallet.Balance -= oldIncomeTransaction.Value;
+                        wallet.DateOfChange = DateTime.UtcNow;
+                    }
+                    else if (wallet.Id == request.updateIncomeDTO.WalletId)
+                    {
+                        wallet.Balance += request.updateIncomeDTO.Value;
+                        wallet.DateOfChange = DateTime.UtcNow;
+                    }
+
+                    updatedWallets.Add(wallet);
+                }
+            }
+            else
+            {
+                foreach (var wallet in wallets)
+                {
+                    if (wallet.Id == request.updateIncomeDTO.WalletId)
+                    {
+                        wallet.Balance = wallet.Balance - oldIncomeTransaction.Value + request.updateIncomeDTO.Value;
+                        wallet.DateOfChange = DateTime.UtcNow;
```

Comment:
instead of making it so complicated, you can make everything much easier and cleaner
```
            public async Task<IncomeTransactionResponse> Handle(UpdateIncomeTransactionCommand request, CancellationToken cancellationToken)
        {
            var filter = Builders<Transaction>.Filter.And(
                Builders<Transaction>.Filter.Eq(t => t.UserId, request.userId),
                Builders<Transaction>.Filter.Eq(t => t.Id, request.updateIncomeDTO.Id),
                Builders<Transaction>.Filter.Eq(t => t.TransactionType, Model.Enums.OperationType.Income));

            var projection = Builders<Transaction>.Projection.Include(t => t.WalletId).Include(t => t.Value);
            var oldIncomeTransaction = (await _dataAccess.FilterBy<Transaction>(filter, projection, cancellationToken)).FirstOrDefault();
            if (oldIncomeTransaction == null) { throw new KeyNotFoundException("Transaction not found"); }
            bool walletChanged = oldIncomeTransaction.WalletId != request.updateIncomeDTO.WalletId;

            var update = Builders<Transaction>.Update
                .Set(t => t.WalletId, request.updateIncomeDTO.WalletId)
                .Set(t => t.CategoryId, request.updateIncomeDTO.CategoryId)
                .Set(t => t.DateOfTransaction, request.updateIncomeDTO.DateOfTransaction)
                .Set(t => t.Value, request.updateIncomeDTO.Value)
                .Set(t => t.Description, request.updateIncomeDTO.Description);

            var updatedIncomeTransaction = await _dataAccess.UpdateOneAsync(filter, update, cancellationToken);
            if (updatedIncomeTransaction == null) { throw new KeyNotFoundException("Transaction not found"); }


            if (walletChanged)
            {
                var a = await _mediator.Send(new ChangeTotalBalanceOfWalletOnDeleteCommand(oldIncomeTransaction), cancellationToken);
                var b = await _mediator.Send(new ChangeTotalBalanceOfWalletCommand(updatedIncomeTransaction), cancellationToken);
                await _mediator.Send(new UpdateWalletDateOfChangeCommand(request.userId, oldIncomeTransaction.WalletId, DateTime.UtcNow), cancellationToken);
            }
            else
            {
                await _mediator.Send(new ChangeTotalBalanceOfWalletOnUpdate(oldIncomeTransaction, updatedIncomeTransaction), cancellationToken);
            }

            await _mediator.Send(new UpdateWalletDateOfChangeCommand(request.userId, updatedIncomeTransaction.WalletId, DateTime.UtcNow), cancellationToken);

            return _mapper.Map<IncomeTransactionResponse>(updatedIncomeTransaction);
        }
```

Reply:
Just use ChangeTotalBalanceOfWalletOnDeleteCommand for 1st wallet and ChangeTotalBalanceOfWalletCommand for 2nd wallet

Reply:
Look at 
```
if(walletChanged)
```

Reply:
@VladislavGromadtsov Did you mean this line of code is duplicated?
`            await _mediator.Send(new UpdateWalletDateOfChangeCommand(request.userId, updatedIncomeTransaction.WalletId, DateTime.UtcNow), cancellationToken);`